libres/lib/enkf/site_config.cpp

```diff
@@ -455,7 +455,7 @@ config_content_type * site_config_alloc_content(
   config_content_type * content = config_parse(
                                         config_parser, site_config_file,
                                         "--", INCLUDE_KEY, DEFINE_KEY, NULL,
-                                        CONFIG_UNRECOGNIZED_WARN, false
+                                        CONFIG_UNRECOGNIZED_WARN, true
```

Comment:
Could you please explain the rationale for why this is resolving the problem at hand?

Reply:
As we understand it, this tells the parser to validate when parsing the site-config. Without this, the required schema items are ignored, and the QUEUE_SYSTEM is not required. It was already supposed to be required, but because of this false, it was not being checked for.

Reply:
https://github.com/equinor/ert/blob/469d62475d08f68e09cf3e674a9d6fb4df1ccff4/libres/lib/config/config_parser.cpp#L646

It is used here.

Reply:
This method:

https://github.com/equinor/ert/blob/469d62475d08f68e09cf3e674a9d6fb4df1ccff4/libres/lib/enkf/queue_config.cpp#L331

adds it as a reqiured item (on line 346), if it is site config. This code was already present.

It is called from 
https://github.com/equinor/ert/blob/469d62475d08f68e09cf3e674a9d6fb4df1ccff4/libres/lib/enkf/site_config.cpp#L373

which is then called from 
https://github.com/equinor/ert/blob/469d62475d08f68e09cf3e674a9d6fb4df1ccff4/libres/lib/enkf/site_config.cpp#L454
and
https://github.com/equinor/ert/blob/469d62475d08f68e09cf3e674a9d6fb4df1ccff4/libres/lib/enkf/model_config.cpp#L735

according to whether it is site config or model config. Thus it will be required in site-config.

Reply:
Thanks 👍🏻 Have you evaluated the other keywords in the `site-config` that is labeled required and the potential consequences if that validation is also ran already for the `site-config`?

Reply:
As far as I can tell, the only other thing that is required anywhere, is NUM_REALIZATIONS, but that is only for model_config, and was already being validated.

We have also ran this against the komodo site-konfig and the default site-config, so it seems to work.

Reply:
Thanks for the clarifications 🙏🏻